generator/src/main/java/com/scottlogic/deg/generator/restrictions/AnyTypeRestriction.java

```diff
@@ -7,7 +7,7 @@
 import java.util.HashSet;
 import java.util.Set;
 
-public class NoTypeRestriction implements TypeRestrictions {
+public class AnyTypeRestriction implements TypeRestrictions {
```

Comment:
Renamed to remove ambiguity and clarify that this type means _any type is permissible_ rather than (potentially) _no types are permissible_